sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/CosmosEncryptionAsyncContainer.java

```diff
@@ -1140,37 +1338,53 @@ public <TContext> Flux<CosmosBulkOperationResponse<TContext>> executeBulkOperati
     public <TContext> Flux<CosmosBulkOperationResponse<TContext>> executeBulkOperations(
         Flux<CosmosItemOperation> operations,
         CosmosBulkExecutionOptions bulkOptions) {
-        if (bulkOptions == null) {
-            bulkOptions = new CosmosBulkExecutionOptions();
-        }
+        final CosmosBulkExecutionOptions cosmosBulkExecutionOptions = Optional.ofNullable(bulkOptions)
+            .orElse(new CosmosBulkExecutionOptions());
 
-        final CosmosBulkExecutionOptions cosmosBulkExecutionOptions = bulkOptions;
         Flux<CosmosItemOperation> operationFlux = operations.flatMap(cosmosItemOperation -> {
-            Mono<CosmosItemOperation> cosmosItemOperationMono;
+            Mono<CosmosItemOperation> cosmosItemOperationMono = null;
             if (cosmosItemOperation.getItem() != null) {
-                ObjectNode objectNode =
-                    EncryptionUtils.getSimpleObjectMapper().valueToTree(cosmosItemOperation.getItem());
-                assert cosmosItemOperation instanceof ItemBulkOperation;
-                cosmosItemOperationMono =
-                    this.encryptionProcessor.encryptObjectNode(objectNode).map(encryptedItem -> new ItemBulkOperation<>(
-                        cosmosItemOperation.getOperationType(),
-                        cosmosItemOperation.getId(),
-                        cosmosItemOperation.getPartitionKeyValue(),
-                        ((ItemBulkOperation<JsonNode, TContext>) cosmosItemOperation).getRequestOptions(),
-                        encryptedItem,
-                        cosmosItemOperation.getContext()
-                    ));
+                cosmosItemOperationMono = this.encryptionProcessor.initEncryptionSettingsIfNotInitializedAsync()
+                    .thenReturn(this.encryptionProcessor.getEncryptionSettings())
+                    .flatMap( encryptionSettings -> {
+                        try {
+                            Field id = cosmosItemOperation.getItem().getClass().getDeclaredField(Constants.PROPERTY_NAME_ID);
```

Comment:
Same here.

Reply:
In the Cosmos Bulk and batch operation, for the create and upsert operations the itemPojo is not null, but the itemId is null when we create the ItemBulkOperation/ItemBatchOperation. I am using that piece of code to get the value for the itemId from the itemPojo. Had a discussion offline as well.